backend/controllers/Event.js

```diff
@@ -83,6 +83,92 @@ const getAllEvents = async (req, res) => {
   }
 };
 
+
+//Add this events to display on MOnth View
+const getAllEventsToday = async (req, res) => {
+  try {
+    const currentDate = new Date(req.query.currentDate);
+    const userId = req.user._id;
+
+    const participation = await EventParticipation.find({
+      userId: userId,
+      status: 'accepted',
+    });
+
+    const eventsIdOfThisUser = participation.map(eventParticipation => {
+      return eventParticipation.eventId;
+    });
+
+    const eventsOfThisUser = await Event.find({
+      _id: { $in: eventsIdOfThisUser },
+      status: { $in: ['overdue', 'ongoing'] },
+    });
+
+    const todayEvents = eventsOfThisUser.filter(event => {
+      const start = new Date(event.start);
+      const end = new Date(event.end);
+      const startOfCurrentDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate(), 0, 0, 0, 1);
+      const endOfCurrentDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate(), 23, 59, 59, 999);
+      console.log(start, end, startOfCurrentDate, endOfCurrentDate, "Date is: ", event.start.toString(), event.end.toString())
+      const minTime = Math.min(start.getTime(), end.getTime(), startOfCurrentDate.getTime(), endOfCurrentDate.getTime())
+      const maxTime = Math.max(start.getTime(), end.getTime(), startOfCurrentDate.getTime(), endOfCurrentDate.getTime())
+      return ((maxTime - minTime) < 24 * 60 * 60 * 1000 + (end.getTime() - start.getTime()));
+    });
```

Comment:
can we abstract this and move to util? It makes the controller file look pretty messy. Also, it seems you try to write the logic yourself. Some library has function like `isAfter` or `isBefore` or some other functions that could you help simplify some of this

Reply:
Roger that, I will try to shorten them